src/vm/arm64/stubs.cpp

```diff
@@ -1995,8 +2010,112 @@ PCODE DynamicHelpers::CreateDictionaryLookupHelper(LoaderAllocator * pAllocator,
 {
     STANDARD_VM_CONTRACT;
 
-    // TODO (NYI)
-    ThrowHR(E_NOTIMPL);
+    // It's available only via the run-time helper function
+    if (pLookup->indirections == CORINFO_USEHELPER)
+    {
+        BEGIN_DYNAMIC_HELPER_EMIT(32);
+
+        // X0 already contains generic context parameter
+        // reuse EmitHelperWithArg for below two operations
+        // X1 <- pLookup->signature
+        // branch to pLookup->helper
+        EmitHelperWithArg(p, pAllocator, (TADDR)pLookup->signature, CEEJitInfo::getHelperFtnStatic(pLookup->helper));
+
+        END_DYNAMIC_HELPER_EMIT();
+    }
+    else
+    {
+        int indirectionsCodeSize = 0;
+        int indirectionsDataSize = 0;
+        for (WORD i = 0; i < pLookup->indirections; i++) {
+            indirectionsCodeSize += (pLookup->offsets[i] > 32760 ? 8 : 4); // if( > 32760) (8 code bytes) else 4 bytes for instruction with offset encoded in instruction
+            indirectionsDataSize += (pLookup->offsets[i] > 32760 ? 4 : 0);
+        }
+
+        int codeSize = indirectionsCodeSize + (pLookup->testForNull ? 44 : 4);
+        if(pLookup->testForNull)
+        {
+            //padding for 8-byte align (required by EmitHelperWithArg)
+            if((codeSize & 0x7) != 0)
```

Comment:
If that is the case, should we check `==` instead of `!=`? I'm seeing `EmitHelperWithArg` has odd number of instructions so 8 byte aligned code size means padding is needed .